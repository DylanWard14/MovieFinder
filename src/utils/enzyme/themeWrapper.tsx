import React from 'react';
import { render, shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import myTheme from '../../theme';

export const renderWrapper = (component: React.ReactNode) => {
    return render(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const shallowWrapper = (component: React.ReactNode) => {
    return shallow(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const mountWrapper = (component: React.ReactNode) => {
    return mount(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};
