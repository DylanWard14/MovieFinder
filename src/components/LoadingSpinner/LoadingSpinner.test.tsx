import React from 'react';
import { render } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import myTheme from '../../theme';
import LoadingSpiner from './LoadingSpinner';

describe('LoadingSpinner', () => {
    it('renders correctly', () => {
        const wrapper = render(
            <ThemeProvider theme={myTheme}>
                <LoadingSpiner />
            </ThemeProvider>,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
