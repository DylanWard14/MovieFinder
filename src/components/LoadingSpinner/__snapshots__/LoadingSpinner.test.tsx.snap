// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner renders correctly 1`] = `
<div>
  <div
    class="LoadingSpinner__Ring-nzk750-0 jkoKXZ"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
